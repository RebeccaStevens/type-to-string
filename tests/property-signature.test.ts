import { describe, it } from "vitest";

import { runTests } from "./helpers";

describe("Property Signatures", () => {
  it.each([
    "type Test = { readonly [key: string]: string };",
    "type Test = { readonly [key: string]: { readonly foo: string; }; };",
    "type Test = { readonly [key: string]: readonly string[] };",
    "type Test = { readonly [key: string]: { foo: string[]; }; };",
    "type Test = { [key: string]: string };",
  ])("%s", (code) => {
    runTests(code);
  });
});
