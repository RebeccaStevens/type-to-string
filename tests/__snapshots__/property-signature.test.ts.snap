// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Property Signatures > type Test = { [key: string]: string }; 1`] = `
{
  "typeAlias": "Test",
  "typeReference": null,
  "typeToString": "Test",
}
`;

exports[`Property Signatures > type Test = { [key: string]: string }; 2`] = `
{
  "typeNodeAlias": "Test",
  "typeNodeAsWritten": "{ [key: string]: string }",
  "typeNodeToString": "{}",
  "typeReferenceNode": null,
}
`;

exports[`Property Signatures > type Test = { readonly [key: string]: { foo: string[]; }; }; 1`] = `
{
  "typeAlias": "Test",
  "typeReference": null,
  "typeToString": "Test",
}
`;

exports[`Property Signatures > type Test = { readonly [key: string]: { foo: string[]; }; }; 2`] = `
{
  "typeNodeAlias": "Test",
  "typeNodeAsWritten": "{ readonly [key: string]: { foo: string[]; }; }",
  "typeNodeToString": "{}",
  "typeReferenceNode": null,
}
`;

exports[`Property Signatures > type Test = { readonly [key: string]: { readonly foo: string; }; }; 1`] = `
{
  "typeAlias": "Test",
  "typeReference": null,
  "typeToString": "Test",
}
`;

exports[`Property Signatures > type Test = { readonly [key: string]: { readonly foo: string; }; }; 2`] = `
{
  "typeNodeAlias": "Test",
  "typeNodeAsWritten": "{ readonly [key: string]: { readonly foo: string; }; }",
  "typeNodeToString": "{}",
  "typeReferenceNode": null,
}
`;

exports[`Property Signatures > type Test = { readonly [key: string]: readonly string[] }; 1`] = `
{
  "typeAlias": "Test",
  "typeReference": null,
  "typeToString": "Test",
}
`;

exports[`Property Signatures > type Test = { readonly [key: string]: readonly string[] }; 2`] = `
{
  "typeNodeAlias": "Test",
  "typeNodeAsWritten": "{ readonly [key: string]: readonly string[] }",
  "typeNodeToString": "{}",
  "typeReferenceNode": null,
}
`;

exports[`Property Signatures > type Test = { readonly [key: string]: string }; 1`] = `
{
  "typeAlias": "Test",
  "typeReference": null,
  "typeToString": "Test",
}
`;

exports[`Property Signatures > type Test = { readonly [key: string]: string }; 2`] = `
{
  "typeNodeAlias": "Test",
  "typeNodeAsWritten": "{ readonly [key: string]: string }",
  "typeNodeToString": "{}",
  "typeReferenceNode": null,
}
`;
